Files/ViewModels/ItemViewModel.cs

```diff
@@ -717,15 +713,22 @@ public async void UnloadItemThumbnail(ListedItem item)
                 {
                     item.LoadUnknownTypeGlyph = true;
                 }
-            }, Windows.System.DispatcherQueuePriority.Low);
-            unloadIconSemaphore.Release();
+            };
+            if (CoreApplication.MainView.DispatcherQueue.HasThreadAccess)
+            {
+                action();
+            }
+            else
+            {
+                await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(action);
+            }
         }
 
-        public async Task LoadItemThumbnail(ListedItem item, uint thumbnailSize = 20)
+        public async Task LoadItemThumbnail(ListedItem item, uint thumbnailSize = 20, IStorageItem matchingStorageItem = null, bool forceReload = false)
```

Comment:
Default arguments used when calling this method from LoadExtendedItemProperties

Reply:
Nice consolidation of the thumbnail logic